src/nvim/tag.c

```diff
@@ -2711,7 +2707,7 @@ expand_tags (
       tagnm[c++] = (t_p.tagkind != NULL && *t_p.tagkind)
                    ? *t_p.tagkind : 'f';
       tagnm[c++] = 0;
-      memmove((*file)[i] + c, t_p.fname, t_p.fname_end - t_p.fname);
+      memmove((*file)[i] + c, t_p.fname, (size_t)(t_p.fname_end - t_p.fname));
```

Comment:
Is it certain that the result of the subtraction will be non-negative? If not then some assert or other check needs to come before this cast.